File: Qytetet_el_juego/p1yp2/QytetetJava/src/modeloqytetet/Tablero.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author chipi
 */
public class Tablero {
    private ArrayList<Casilla> casillas; //MAX_CASILLAS, ordenado por numeroCasilla
    private Casilla carcel;

    public Tablero() {
        inicializar();
    }
    /*
    boolean esCasillaCarcel(int numeroCasilla){
        return true
    }
    */

    public Casilla getCarcel() {
        return carcel;
    }
    
    /*
    Casilla obtenerCasillaNumero(int numeroCasilla){
        return null;
    }
    
    Casilla obtenerNuevaCasilla(Casilla casilla, int desplazamiento){
        return null;
    }
    */

    @Override
    public String toString() {
        return "Tablero{" + "casillas=" + casillas + ", carcel=" + carcel + '}';
    }
    
    private void inicializar(){
        this.casillas = new ArrayList();
        
        Casilla calle = new Casilla(0,0,TipoCasilla.SALIDA);
        casillas.add(calle);
        
        TituloPropiedad tp = new TituloPropiedad("Tomate",false,50,10,150,250);
        calle= new Casilla(1,250,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Estrella",false,50,12,250,250);
        calle = new Casilla(2,250,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(3,0,TipoCasilla.SORPRESA);
        casillas.add(calle);
        
        carcel = new Casilla(4,0,TipoCasilla.CARCEL); ///////////////////////CARCEL////////
        casillas.add(carcel);
        
        tp = new TituloPropiedad("Pimiento",false,50,16,350,350);
        calle = new Casilla(5,350,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Melon",false,50,20,450,350);
        calle = new Casilla(6,350,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Sandia",false,50,18,550,450);
        calle = new Casilla(7,450,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(8,0,TipoCasilla.IMPUESTO);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Melocoton",false,50,14,650,450);
        calle = new Casilla(9,450,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(10,0,TipoCasilla.SORPRESA);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Patata",false,100,18,750,500);
        calle = new Casilla(11,500,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(12,0,TipoCasilla.JUEZ);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Ciruela",false,100,10,800,500);
        calle = new Casilla(13,500,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Almibar",false,100,12,850,600);
        calle = new Casilla(14,600,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(15,0,TipoCasilla.SORPRESA);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Botón",false,100,18,900,650);
        calle = new Casilla(16,450,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Silla",false,100,14,950,700);
        calle = new Casilla(17,450,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        calle = new Casilla(18,0,TipoCasilla.PARKING);
        casillas.add(calle);
        
        tp = new TituloPropiedad("Aprobado",false,100,20,1000,750);
        calle = new Casilla(19,450,0,0,TipoCasilla.CALLE,tp);
        casillas.add(calle);
        
        
        /*  public TituloPropiedad(String nombre, boolean hipotecada, int alquilerBase, float factorRevalorizcion, int hipotecaBase, int precioEdificar)
            public Casilla(int numeroCasilla, int coste, int numHoteles, int numCasas, TipoCasilla tipo, TituloPropiedad titulo) 
            public Casilla(int numeroCasilla, int coste, TipoCasilla tipo) {

        */
    }
    
}

File: Examenes/examen1/QytetetRuby/lib/tablero.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tipo_casilla"
require_relative "casilla"
require_relative "titulo_propiedad"

module ModeloQytetet
  class Tablero
    attr_accessor :carcel, :casillas
    
    #@carcel # casilla
    
    def initialize
      inicializar
    end
    
    def to_s
      "Tablero: \n
        casillas: #{@casillas} \n 
        carcel: #{@carcel} \n "
    end
    
    
    def esCasillaCarcel(numeroCasilla)#boolean esCasillaCarcel(int numeroCasilla)
      raise ArgumentError,'metodo sin implementar'
    end
    
    def obtenerCasillaNumero(numeroCasilla)#Casilla obtenerCasillaNumero(int numeroCasilla)
      return @casillas[numeroCasilla]
    end
    
    def obtenerNuevaCasilla(casilla, desplazamiento)#Casilla obtenerNuevaCasilla(Casilla casilla, int desplazamiento)
      raise ArgumentError,'metodo sin implementar'
    end
    
    private
    
    def inicializar 
      @casillas = Array.new
      
      calle = Casilla.crear_casillas_especiales(0,TipoCasilla::SALIDA)
      @casillas << calle
      
      tomate = TituloPropiedad.new("Tomate",false,50,10,150,250)
      calle = Casilla.crear_casillas_calle(1,250,tomate)
     puts 'pruebasssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss'
      puts calle.coste
      puts tomate.casilla
      @casillas << calle
      
      estrella = TituloPropiedad.new("Estrella",false,50,12,250,250)
      calle = Casilla.crear_casillas_calle(2,250,estrella)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(3,TipoCasilla::SORPRESA)
      @casillas << calle
      
      @carcel = Casilla.crear_casillas_especiales(4,TipoCasilla::CARCEL) ############### CARCEL #################
      @casillas << @carcel
      
      pimiento = TituloPropiedad.new("Pimiento",false,50,16,350,350)
      calle = Casilla.crear_casillas_calle(5,350,pimiento)
      @casillas << calle
      
      melon = TituloPropiedad.new("Melon",false,50,20,450,350)
      calle = Casilla.crear_casillas_calle(6,350,melon)
      @casillas << calle
      
      sandia = TituloPropiedad.new("Sandia",false,50,18,550,450)
      calle = Casilla.crear_casillas_calle(7,450,sandia)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(8,TipoCasilla::IMPUESTO)
      @casillas << calle
      
      melocoton = TituloPropiedad.new("Melocoton",false,50,14,650,450)
      calle = Casilla.crear_casillas_calle(9,450,melocoton)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(10,TipoCasilla::SORPRESA)
      @casillas << calle
      
      patata = TituloPropiedad.new("Patata",false,100,18,750,500)
      calle = Casilla.crear_casillas_calle(11,500,patata)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(12,TipoCasilla::JUEZ)
      @casillas << calle
      
      ciruela = TituloPropiedad.new("Ciruela",false,100,10,800,500)
      calle = Casilla.crear_casillas_calle(13,500,ciruela)
      @casillas << calle
      
      almibar = TituloPropiedad.new("Almibar",false,100,12,850,600)
      calle = Casilla.crear_casillas_calle(14,600,almibar)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(15,TipoCasilla::SORPRESA)
      @casillas << calle
      
      boton = TituloPropiedad.new("Boton",false,100,18,900,650)
      calle = Casilla.crear_casillas_calle(16,450,boton)
      @casillas << calle
      
      silla = TituloPropiedad.new("Silla",false,100,14,950,700)
      calle = Casilla.crear_casillas_calle(17,450,silla)
      @casillas << calle
      
      calle = Casilla.crear_casillas_especiales(18,TipoCasilla::PARKING)
      @casillas << calle
      
      aprobado = TituloPropiedad.new("Aprobado",false,100,20,1000,750)
      calle = Casilla.crear_casillas_calle(19,450,aprobado)
      @casillas << calle
    end  
  end
end 